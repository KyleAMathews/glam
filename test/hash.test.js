/* eslint-env jest */
const hashArray = require('../src/hash').default

describe('hash', () => {
  test('str length: 1', () => {
    const cls = '1'
    const vars = ['a', 'b', 'c', 'd']
    expect(hashArray([cls, ...vars])).toMatchSnapshot()
  })

  test('str length: 2', () => {
    const cls = '12'
    const vars = ['a', 'b', 'c', 'd']
    expect(hashArray([cls, ...vars])).toMatchSnapshot()
  })

  test('str length: 3', () => {
    const cls = '123'
    const vars = ['a', 'b', 'c', 'd']
    expect(hashArray([cls, ...vars])).toMatchSnapshot()
  })

  test('str length: >= 4', () => {
    const cls = '1234'
    const vars = ['a', 'b', 'c', 'd']
    expect(hashArray([cls, ...vars])).toMatchSnapshot()
  })
})
