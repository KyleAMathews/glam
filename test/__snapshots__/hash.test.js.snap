// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hash 1 1`] = `"1yunt80"`;

exports[`hash 2 1`] = `"9p294j"`;

exports[`hash 3 1`] = `"527z82"`;

exports[`hash 4 1`] = `"1bpnzwg"`;

exports[`hash str length: >= 4 1`] = `"1bpnzwg"`;

exports[`hash str length: 1 1`] = `"1yunt80"`;

exports[`hash str length: 2 1`] = `"9p294j"`;

exports[`hash str length: 3 1`] = `"527z82"`;
